packages/aws-cdk-lib/aws-apigatewayv2/lib/http/route.ts

```diff
@@ -0,0 +1,263 @@
+   */
+  readonly authorizationScopes?: string[];
+}
+
+/**
+ * Supported Route Authorizer types
+ */
+enum HttpRouteAuthorizationType {
+  /** AWS IAM */
+  AWS_IAM = 'AWS_IAM',
+
+  /** JSON Web Tokens */
+  JWT = 'JWT',
+
+  /** Lambda Authorizer */
+  CUSTOM = 'CUSTOM',
+
+  /** No authorizer */
+  NONE = 'NONE'
+}
+
+/**
+ * Route class that creates the Route for API Gateway HTTP API
+ * @resource AWS::ApiGatewayV2::Route
+ */
+export class HttpRoute extends Resource implements IHttpRoute {
+  public readonly routeId: string;
+  public readonly httpApi: IHttpApi;
+  public readonly path?: string;
+  public readonly routeArn: string;
+
+  private readonly method: HttpMethod;
+  private readonly authBindResult?: HttpRouteAuthorizerConfig;
+
+  constructor(scope: Construct, id: string, props: HttpRouteProps) {
+    super(scope, id);
+
+    this.httpApi = props.httpApi;
+    this.path = props.routeKey.path;
+    this.method = props.routeKey.method;
+    this.routeArn = this.produceRouteArn(props.routeKey.method);
+
+    const config = props.integration._bindToRoute({
+      route: this,
+      scope: this,
+    });
+
+    const authorizer = props.authorizer ?? this.httpApi.defaultAuthorizer;
+    this.authBindResult = authorizer?.bind({
+      route: this,
+      scope: this.httpApi instanceof Construct ? this.httpApi : this, // scope under the API if it's not imported
+    });
+
+    if (this.authBindResult && !(this.authBindResult.authorizationType in HttpRouteAuthorizationType)) {
+      throw new Error(`authorizationType should either be AWS_IAM, JWT, CUSTOM, or NONE but was '${this.authBindResult.authorizationType}'`);
+    }
+
+    let authorizationScopes = this.authBindResult?.authorizationScopes;
+
+    if (this.authBindResult && (props.authorizationScopes || this.httpApi.defaultAuthorizationScopes)) {
+      authorizationScopes = Array.from(new Set([
+        ...authorizationScopes ?? [],
+        ...props.authorizationScopes ?? this.httpApi.defaultAuthorizationScopes ?? [],
+      ]));
+    }
+
+    if (authorizationScopes?.length === 0) {
+      authorizationScopes = undefined;
+    }
+
+    const routeProps: CfnRouteProps = {
+      apiId: props.httpApi.apiId,
+      routeKey: props.routeKey.key,
+      target: `integrations/${config.integrationId}`,
+      authorizerId: this.authBindResult?.authorizerId,
+      authorizationType: this.authBindResult?.authorizationType ?? 'NONE',
+      authorizationScopes,
+    };
+
+    const route = new CfnRoute(this, 'Resource', routeProps);
+    this.routeId = route.ref;
+  }
+
+  private produceRouteArn(httpMethod: HttpMethod): string {
+    const stage = '*';
+    const iamHttpMethod = httpMethod === HttpMethod.ANY ? '*' : httpMethod;
+    const path = this.path ?? '/';
+    // When the user has provided a path with path variables, we replace the
+    // path variable and all that follows with a wildcard.
+    const iamPath = path.replace(/\{.*?\}.*/, '*');
```

Comment:
Unsure how to rewrite this regex, so I added the following to throw an error if the path string is too large:

```ts
// When the user has provided a path with path variables, we replace the
// path variable and all that follows with a wildcard.
if (path.length > 1000) {
  throw new Error(`Path is too long: ${path}`);
};
const iamPath = path.replace(/\{.*?\}.*/, '*');
```

Reply:
The last example from the [GitHub code scanning comment](https://github.com/aws/aws-cdk/security/code-scanning/115) explains:

> **Example**
> Sometimes it is unclear how a regular expression can be rewritten to avoid the problem. In such cases, it often suffices to limit the length of the input string. For instance, the following regular expression is used to match numbers, and on some non-number inputs it can have quadratic time complexity:
> ```python
> /^(\+|-)?(\d+|(\d*\.\d*))?(E|e)?([-+])?(\d+)?$/.test(str) // BAD
> ```
> It is not immediately obvious how to rewrite this regular expression to avoid the problem. However, you can mitigate performance issues by limiting the length to 1000 characters, which will always finish in a reasonable amount of time.
> 
> ```python
> if (str.length > 1000) {
>     throw new Error("Input too long");
> }
> 
> /^(\+|-)?(\d+|(\d*\.\d*))?(E|e)?([-+])?(\d+)?$/.test(str)
> ```